### ISSUE TITLE
[Question/Clarification/Possible Bug] Soft deletion does not happen when passing in nil struct

### ISSUE THREAD

**Issue by ernsheong** [go-pg/pg#1232]
I have a model Foo with `deletedAt`:

```go
type Foo struct {
  ...
  DeletedAt   time.Time         `json:"deletedAt" pg:",soft_delete"`
}
```

```go
db.Model((*Foo)(nil)).Where("attr = ?", attr).Delete()
```

results in:

```sql
DELETE FROM "foos" AS "foo" WHERE ((attr = 1)) AND "foo"."deleted_at" IS NOT NULL
```

Note that:
1. Soft deletion did not take place (unexpected), and it tried to hard delete
2. Model hooks not called

On the other hand:

```go
db.Model(&Foo{}).Where("attr = ?", attr).Delete()
```

will invoke model hooks, even though the model passed in is a dummy, resulting in the SQL:

```sql
UPDATE "foos" AS "foo" SET "deleted_at" = '2019-06-03 06:35:07.570936+00:00:00' WHERE ((attr = 1)) AND "foo"."deleted_at" IS NULL
```

Note that:
1. Soft deletion took place as expected
2. Model hooks were called (even though the struct is uninitialized with zero values, does it even make sense to do so?)

What is the difference between passing in `(*Foo)(nil)` and `&Foo{}`? Why is the first not soft deleting?

Basically, I want to be able to bulk soft delete without having to retrieve all the models first. Is that possible?

Version: 8.0.4

**Comment by vmihailenco** [comment]
This is an overlook that should be fixed in master. Thanks for the report.
